### DISCUSSION TITLE
Printer-friendly Pages with CSS

### DISCUSSION THREAD

**Discussion by rhodelta66** [sqlpage/SQLPage#742] [Ideas]
We are very happy with the _md text options.  We use it frequently to create online manuals with code examples for our [tool.](https://www.t-sql.app)

The SQLPage -pages look amazing, but the code fragments in the browser often have a scroll-bar due to the size of the ```code``` fragments. 

It would be great is the CSS could be by default Printer-friendly sio that everything in the code blocks will be printed. We can add code to the shell component, but maybe havng it solved by default would be better for everyone.

**Comment by lovasoa** [comment]
That would be very welcome indeed! Can you make a PR with changes to the SQLPage css?

**Comment by rhodelta66** [comment]
Adding this tot the css will help everyone for this situation:

@media print {
    pre:has(code) {
        max-height: none !important;
    }
}

**Comment by rhodelta66** [comment]
Bonjour Ophir,

Ce dimanche, j'ai cloné SQLPage en local et j'ai même réussi à compiler la
version v0.32 (après avoir installé Rust), donc je me sens comme un pro
maintenant, haha.

Je ne sais pas où je devrais modifier le code pour que le CSS par défaut de
SQLPage inclue également le "CSS adapté à l'impression". J'ai remarqué que
vous ne l'aviez pas encore ajouté à la documentation principale de SQLPage,
donc quand les gens voudront imprimer les pages d'exemple, ils manqueront
inutilement certains détails du code.

Si vous pouvez m'indiquer où effectuer les modifications, je le ferai en
local, je le testerai, et si tout fonctionne bien, je vous enverrai une
pull request.

Merci pour tout votre excellent travail !

Joyeux Noël !

Rick Hoek

(Thanks to AI translation, parseque moi je ne parle pas si bon Francais ;)





Op di 17 dec 2024 om 14:01 schreef Ophir LOJKINE ***@***.***>:

> That would be very welcome indeed! Can you make a PR with changes to the
> SQLPage css?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/sqlpage/SQLPage/discussions/742#discussioncomment-11593092>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHS2S7BVTXRX5DXWZDML6632GAOCBAVCNFSM6AAAAABTYJMB52VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNJZGMYDSMQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

Met vriendelijke groet,



Rick Hoek

Directeur



Tracy Data Solutions B.V.*|* ABC Westland 666/670 *|* 2685 DH Poeldijk *|*
 Netherlands

*Ph:* +31 10 744 18 13 *|* *Mob:* +31 6 53 19 74 06 *|* *E: ***@***.***
*|* *I:* www.tracy.nu

Aan dit bericht kunnen geen rechten worden ontleend.

**Reply by lovasoa** [reply]
Merci pour ce gentil message :)

This info may be useful to others, so I'll switch back to english.

The main css file that is included in all pages in `sqlpage/sqlage.css`; you'll find it here: https://github.com/sqlpage/SQLPage/blob/main/sqlpage/sqlpage.css
